src/main/java/io/vertx/core/parsetools/impl/JsonParserImpl.java

```diff
@@ -221,6 +232,16 @@ public void end() {
   private void checkPending() {
     if (!emitting) {
       emitting = true;
+      while (demand > 0L && !pendingEvents.isEmpty()) {
+        if (demand != Long.MAX_VALUE) {
+          demand--;
+        }
+        JsonEvent jsonEvent = pendingEvents.removeFirst();
+        Handler<JsonEvent> handler = this.eventHandler;
+        if (handler != null) {
+          handler.handle(jsonEvent);
+        }
+      }
```

Comment:
The while loop should be inside the try clause. If the eventHandler throws an exception we want `finally` `emitting = false;`

Reply:
Right